great_expectations/profile/json_schema_profiler.py

```diff
@@ -185,12 +255,27 @@ def _create_string_length_expectation(
         self, key: str, details: dict
     ) -> Optional[ExpectationConfiguration]:
         """https://json-schema.org/understanding-json-schema/reference/string.html#length"""
-        type_ = details.get("type", None)
-        minimum = details.get("minLength", None)
-        maximum = details.get("maxLength", None)
+        object_types = self._get_object_types(details=details)
 
-        if type_ != JsonSchemaTypes.STRING.value:
+        if JsonSchemaTypes.STRING.value not in object_types:
             return None
+
+        type_ = details.get("type", None)
+        any_of = details.get("anyOf", None)
+
+        if type_:
+            minimum = details.get("minLength", None)
+            maximum = details.get("maxLength", None)
+        elif any_of:
+            for item in any_of:
+                item_type = item.get("type", None)
+                if item_type == JsonSchemaTypes.STRING.value:
+                    minimum = item.get("minLength", None)
+                    maximum = item.get("maxLength", None)
+                    break
+        else:
+            return None
```

Comment:
```suggestion
```

Reply:
Good point, that is a the right way to guard and exit early. Change made.